pkg/result/ignore.go

```diff
@@ -218,16 +219,15 @@ func ParseIgnoreFile(ctx context.Context, ignoreFile string) (IgnoreConfig, erro
 
 func parseIgnoreYAML(ignoreFile string) (IgnoreConfig, error) {
 	// Read .trivyignore.yaml
-	f, err := os.Open(ignoreFile)
+	b, err := os.ReadFile(ignoreFile)
 	if err != nil {
 		return IgnoreConfig{}, xerrors.Errorf("file open error: %w", err)
 	}
-	defer f.Close()
 	log.Debug("Found an ignore yaml", log.FilePath(ignoreFile))
 
 	// Parse the YAML content
 	var ignoreConfig IgnoreConfig
-	if err = yaml.NewDecoder(f).Decode(&ignoreConfig); err != nil {
+	if err = yaml.Unmarshal(b, &ignoreConfig); err != nil {
```

Comment:
`Decode` is better in terms of memory efficiency. We should leave a comment explaining why we use `Unmarshal` here. Otherwise, someone may change it back in the future for performance reasons.

Reply:
added https://github.com/aquasecurity/trivy/pull/7962/commits/b304f648ee303f3ec6f5e9d592a4c96a0a94dd0e